docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -0,0 +1,127 @@
+
+Integration of the quote-ex library depends on the installation of the Intel® SGX SDK quote-ex library package and its dependencies, as well as proper configuration of the components and their access to dependent backend services. Details for the quote-ex library installation and configuration are outside the scope of this document.
+
+# Specification
+
+## Existing OE SDK Implementation
+
+### Evidence Format Enumeration and Plugin Registration
+
+Existing OE SDK implementation based on the DCAP library only supports generation of evidence in a single SGX ECDSA-p256 format, so there is no need for enumeration of supported evidence formats. As implemented in code file `enclave/sgx/attester.c`, a single attester plugin is created for the SGX ECDSA-p256 evidence format.
+- Note: in the current OE SDK implementation, the UUID for the ECDSA-p256 evidence format is still called `OE_SGX_PLUGIN_UUID`, which is the same as `OE_SGX_ECDSA_P256_PLUGIN_UUID`.
+
+### Implementation of OE SDK API `oe_get_evidence()`
+
+The implementation of OE SDK API `oe_get_evidence()`, in code file `common/attest_plugin.c`, searches for an attester plugin that supports the requested evidence format, and invokes the `get_evidence()` entry point of the selected plugin.
+
+The SGX ECDSA-p256 attester plugin is implemented in code file `enclave/sgx/attester.c` and other relevant enclave-side and host-side code files, called enclave-side and host-side plugin libraries in this document. The enclave-side plugin library interacts with the host-side plugin library via OCALLs defined in interface definition file `common/sgx/sgx.edl`. For SGX ECDSA-p256 evidence generation, there are 2 OCALLs:
+
+- `oe_get_qetarget_info_ocall(sgx_target_info_t* target_info)`
+    - Return the SGX Quoting Enclave (QE) target information.
+- `oe_get_quote_ocall(const sgx_report_t* sgx_report, void* quote, size_t quote_size, size_t* quote_size_out)`
+    - Generate an ECDSA-p256 quote and return in the caller-supplied buffer, or return the needed buffer size if the supplied buffer is missing or not large enough.
+
+Since only a single evidence format is supported and this format does not require any optional parameter, these OCALLs pass neither the evidence format ID nor optional parameter.
+
+The host-side plugin library implements the OCALLs, as in code file `host/sgx/ocalls.c` and other relevant code files. As defined in the main `cmake` configuration file `CMakeLists.txt` in the OE SDK top directory, the DCAP library is linked to the OE SDK host-side plugin library. The DCAP library provides following 3 API functions in support of the above two OCALLs, as defined in its [header file](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/ql/inc/sgx_dcap_ql_wrapper.h).
+
+- `sgx_qe_get_target_info(sgx_target_info_t *p_qe_target_info)`
+    - Return the SGX Quoting Enclave (QE) target information, for application enclave to generate its SGX report.
```

Comment:
grammar: "for application" -> "for the application"

Reply:
Thanks!